corehq/apps/users/models.py

```diff
@@ -2088,19 +2088,25 @@ def get_usercase_id(self):
     def update_device_id_last_used(self, device_id, when=None):
         """
         Sets the last_used date for the device to be the current time
-
         Does NOT save the user object.
+
+        :returns: True if user was updated and needs to be saved
         """
         when = when or datetime.utcnow()
+
         for user_device_id_last_used in self.devices:
             if user_device_id_last_used.device_id == device_id:
-                user_device_id_last_used.last_used = when
-                break
+                if when.date() > user_device_id_last_used.last_used.date():
```

Comment:
What is the advantage of doing this over storing just the date and using `date.today()` (keeping everything in dates instead of always converting from datetimes)